samples/sensor/sensor_shell/sample.yaml

```diff
@@ -11,3 +11,16 @@ tests:
     harness: keyboard
     min_ram: 20
     min_flash: 33
+  sample.sensor.shell.pytest:
+    filter: CONFIG_SERIAL and dt_chosen_enabled("zephyr,shell-uart")
+    min_ram: 40
+    harness: pytest
+    extra_configs:
+      - arch:posix:CONFIG_NATIVE_UART_0_ON_STDINOUT=y
+      - CONFIG_SAMPLES_SENSOR_SHELL_FAKE_SENSOR=y
+    integration_platforms:
+      - qemu_riscv64
+      - native_posix
```

Comment:
Any reason this is not `native_sim`?

Reply:
`native_posix` is what I always use for simple software stuff like this one, `qemu_riscv64` is what I use when `native` doesn't compile (MacOS), `riscv` is the arch that I work with on a daily basis. Just out-of-habit I guess

Reply:
Updated to use `native_sim`, since that seems to be the direction

Reply:
`native_sim` seems to go bonkers in CI..

Reply:
It went bonkers after #71093, probably not `native_sim` related

Reply:
The sensor shell now stuck in an infinite loop when doing `sensor get <sensor> <channel>`, @teburd do you have any idea why?

Reply:
doh, thanks for finding and fixing, I think this is what @ubieda had fixed as well